src/core/data/grid/gridlayoutimplyee.hpp

```diff
@@ -508,6 +508,96 @@ namespace core
 
 
 
+        NO_DISCARD auto static constexpr BxToMoments()
+        {
+            // Bx is primal dual dual
+            // moments are primal primal primal
+            // operation is thus Pdd to Ppp
+            [[maybe_unused]] auto constexpr iShift = dualToPrimal();
+
+            if constexpr (dimension == 1)
+            {
+                constexpr WeightPoint<dimension> P1{Point<int, dimension>{0}, 1.};
+                return std::array<WeightPoint<dimension>, 1>{P1};
+            }
+            if constexpr (dimension == 2)
+            {
+                constexpr WeightPoint<dimension> P1{Point<int, dimension>{0, 0}, 0.5};
+                constexpr WeightPoint<dimension> P2{Point<int, dimension>{0, iShift}, 0.5};
+                return std::array<WeightPoint<dimension>, 2>{P1, P2};
+            }
+            else if constexpr (dimension == 3)
```

Comment:
I think this cannot be right, in 3D there are 4 faces around a corner node, like there are 4 faces of Bz in 2D around a corner node. This is the same comment for Bx, By and Bz